From 1a0e71a04479dbd6ef08f46b4911ddc64b47347d Mon Sep 17 00:00:00 2001
From: Icecream95 <ixn@disroot.org>
Date: Tue, 6 Dec 2022 16:52:06 +1300
Subject: [PATCH 5/5] drm/rockchip: Re-add implicit fencing support for planes

Similar to 63d5e06aa381 ("drm/rockchip: Add implicit fencing support
for planes") but after the code was removed and VOP2 was added in the
downstream kernel.

Note that upstream VOP2 does not have this line as it was added since
29a840842743 ("drm/<driver>: drm_gem_plane_helper_prepare_fb is now
the default") removed the need.

---
 drivers/gpu/drm/rockchip/rockchip_drm_vop2.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
index df4a3291d3da..f9675d9320d4 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
@@ -10,6 +10,7 @@
 #include <drm/drm_crtc_helper.h>
 #include <drm/drm_debugfs.h>
 #include <drm/drm_flip_work.h>
+#include <drm/drm_gem_framebuffer_helper.h>
 #include <drm/drm_fourcc.h>
 #include <drm/drm_plane_helper.h>
 #include <drm/drm_probe_helper.h>
@@ -4622,6 +4623,7 @@ static const struct drm_plane_helper_funcs vop2_plane_helper_funcs = {
 	.atomic_check = vop2_plane_atomic_check,
 	.atomic_update = vop2_plane_atomic_update,
 	.atomic_disable = vop2_plane_atomic_disable,
+	.prepare_fb = drm_gem_fb_prepare_fb,
 };
 
 /**
-- 
2.38.1

